web/pdf_attachment_viewer.js

```diff
@@ -89,6 +89,34 @@ var PDFAttachmentViewer = (function PDFAttachmentViewerClosure() {
     /**
      * @private
      */
+    _bindPdfLink:
+        function PDFAttachmentViewer_bindPdfLink(button, content, filename) {
+      var blobUrl;
+      button.onclick = function() {
+        if (!blobUrl) {
+          blobUrl = pdfjsLib.createObjectURL(content, 'application/pdf') +
+            '#' + filename;
+        }
+        var viewerUrl;
+        if (typeof PDFJSDev === 'undefined' || PDFJSDev.test('GENERIC')) {
+          // The current URL is the viewer, let's use it and append the file.
+          viewerUrl = '?file=' + encodeURIComponent(blobUrl);
+        } else if (PDFJSDev.test('CHROME')) {
+          // eslint-disable-next-line no-undef
+          viewerUrl = chrome.runtime.getURL('/content/web/viewer.html') +
+            '?file=' + encodeURIComponent(blobUrl);
+        } else {
+          // Let Firefox's content handler catch the URL and display the PDF.
+          viewerUrl = blobUrl;
```

Comment:
In the Firefox addon, the `#` used above causes the viewer to try and goto a named destination equal to the `filename`. Most likely such a destination won't exist, but it'd be great to try and avoid this altogether.